src/clear/model/json_deserialize.cr

```diff
@@ -0,0 +1,71 @@
+macro columns_to_instance_vars
+  struct Assigner
+    include JSON::Serializable
+
+    {% for name, settings in COLUMNS %}
+      getter {{name.id}} : {{settings[:type]}}?
+    {% end %}
+
+    def create
+      assign_columns({{@type}}.new) # Loop through instance variables and assign to the newly created orm instance
+    end
+
+    def update(model)
+      assign_columns(model) # Loop through instance variables and assign to the orm instance you are updating
+    end
+
+    macro finished
+      protected def assign_columns(model)
+        {% for name, settings in COLUMNS %}
+          model.{{name.id}} = @{{name.id}}.not_nil! unless @{{name.id}}.nil?
+        {% end %}
+
+        model
+      end
+    end
+  end
+
+  def self.from_json(string_or_io : String | IO)
+    Assigner.from_json(string_or_io).create
+  end
+
+  def self.create_from_json(string_or_io : String | IO)
+    mdl = self.from_json(string_or_io)
+    mdl.save
+    mdl
+  end
+
+  def self.create_from_json!(string_or_io : String | IO)
+    self.from_json(string_or_io).save!
+  end
+
+  def set_from_json(string_or_io : String | IO)
+    Assigner.from_json(string_or_io).update(self)
+  end
+
+  def update_from_json(string_or_io : String | IO)
+    mdl = set_from_json(string_or_io)
+    mdl.save
+    mdl
```

Comment:
```suggestion
```

Reply:
I have to return mdl here otherwise it would return a Boolean, see line 94 https://github.com/anykeyh/clear/blob/5e233e549894010ed68ca949edf8f532d4fa086f/src/clear/model/modules/has_saving.cr

Reply:
I'm just mirroring this

```
      def self.create(**args) : self
        mdl = build(**args)
        mdl.save
        mdl
      end
```

https://github.com/anykeyh/clear/blob/5e233e549894010ed68ca949edf8f532d4fa086f/src/clear/model/modules/class_methods.cr